CMSgov/design-system
# [WNMGDS 968] convert button component to typescript

## Summary
Converting `<Button>` from JS to TS.

### Changed
- Changed file extension from `.jsx` to `.tsx` in docs
- Changed file extension from `.jsx` to `.tsx` in unit test, removed `.jsx` from `<Button>` import
- Changed file extension from `.jsx` to `.tsx` in snapshot test
- `<Button>` types imported from component file instead of type definition file
- Added `onClick` to `OmitProps` as our ButtonProps interface overrides the `onClick` prop defined on the button element

### Removed
- Button type definition file

## How to test
Navigate to [Button component](http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-968/convert-button-to-typescript/) and confirm there are no console errors and documentation renders as expected.
